crates/pop-parachains/src/bench/mod.rs

```diff
@@ -187,31 +170,51 @@ fn process_pallet_extrinsics(output_file: PathBuf) -> anyhow::Result<PalletExtri
 	Ok(registry)
 }
 
-/// Run command for benchmarking with a provided binary.
+/// Run command for benchmarking with a provided `frame-omni-bencher` binary.
 ///
 /// # Arguments
 /// * `binary_path` - Path to the binary to run.
-/// * `output` - Output file to write the benchmark results to.
+/// * `command` - Command to run. `frame-omni-bencher` only supports `pallet` and `overhead`.
 /// * `args` - Additional arguments to pass to the binary.
-pub async fn run_benchmarking_with_binary(
+/// * `log_enabled` - Whether to enable logging.
+pub fn generate_omni_bencher_benchmarks(
 	binary_path: &Path,
-	output: Option<&File>,
-	args: Vec<&str>,
-) -> anyhow::Result<Child> {
-	let mut command = Command::new(binary_path);
-	let env = std::env::var("RUST_LOG").unwrap_or_default();
-	command.env("RUST_LOG", "none");
-	command.args(["v1", "benchmark", "pallet"]);
-	for arg in args {
-		command.arg(arg);
-	}
-	if let Some(output) = output {
-		command.stdout(Stdio::from(output.try_clone()?));
-		command.stderr(Stdio::from(output.try_clone()?));
+	command: &str,
+	args: Vec<String>,
+	log_enabled: bool,
+) -> anyhow::Result<String> {
+	let stdout_file = NamedTempFile::new()?;
+	let stdout_path = stdout_file.path().to_owned();
+
+	let stderror_file = NamedTempFile::new()?;
+	let stderror_path = stderror_file.path().to_owned();
+
+	let mut cmd_args = vec!["v1".to_string(), "benchmark".to_string(), command.to_string()];
+	cmd_args.append(&mut args.clone());
+
+	let cmd = cmd(binary_path, cmd_args)
+		.env("RUST_LOG", if log_enabled { "info" } else { "none" })
+		.stderr_path(&stderror_path)
+		.stdout_path(&stdout_path);
+
+	if let Err(e) = cmd.run() {
+		let mut error_output = String::new();
+		std::fs::File::open(&stderror_path)
+			.unwrap()
+			.read_to_string(&mut error_output)
+			.unwrap();
+		return Err(anyhow::anyhow!(
+			"Failed to run benchmarking: {}",
+			if error_output.is_empty() { e.to_string() } else { error_output }.trim()
+		));
 	}
-	let child = command.spawn()?;
-	command.env("RUST_LOG", env);
-	Ok(child)
+
+	let mut stdout_output = String::new();
+	std::fs::File::open(&stdout_path)
+		.unwrap()
+		.read_to_string(&mut stdout_output)
```

Comment:
```suggestion
		.read_to_string(&mut stdout_output)?
```
As above remove unwrap()